lib/xray_creator.py

```diff
@@ -47,10 +47,7 @@ def _initialize_books(self):
         book_lookup = {}
         for book in self.books_not_failing():
             self._total_not_failing += 1
-            if book_lookup.has_key(book.title):
```

Comment:
self.books_not_failing() is a generator. I can't use defaultdict with it :(

On Thu, Jun 23, 2016 at 11:56 AM Anthony Toole <notifications@github.com>
wrote:

> In lib/xray_creator.py
> <https://github.com/szarroug3/X-Ray_Calibre_Plugin/pull/52#discussion_r68271999>
> :
>
> > @@ -47,10 +47,7 @@ def _initialize_books(self):
> >          book_lookup = {}
> >          for book in self.books_not_failing():
> >              self._total_not_failing += 1
> > -            if book_lookup.has_key(book.title):
>
> if you need to do this again, then defaultdict is your friend:
>
> book_lookup = defaultdict(list)
> for ...
>   book_lookup[book.title].append(book)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/szarroug3/X-Ray_Calibre_Plugin/pull/52/files/09c125a697b713dc4d4ddc7ffa0dec29542c36bf#r68271999>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AIEjUXqJrHXVI3kjo_SwOvcPwhvCBDe5ks5qOrqogaJpZM4I8-nd>
> .
>